torchgeometry/imgwarp.py

```diff
@@ -135,9 +142,9 @@ def warp_affine(src, M, dsize, flags='bilinear', border_mode=None,
         raise ValueError("Input M must be a Bx2x3 tensor. Got {}"
                          .format(src.shape))
     # we generate a 3x3 transformation matrix from 2x3 affine
-    M_3x3 = torch.eye(3, device=src.device, dtype=src.dtype)
-    M_3x3 = torch.unsqueeze(M_3x3, dim=0).repeat(src.shape[0], 1, 1)
-    M_3x3[..., :2, :3] = M
+    M_3x3 = F.pad(M, (0, 0, 0, 1, 0, 0))
```

Comment:
sweet! :D

Reply:
Again, JIT cannot replace a tensor into another. If we want to replace `M` into `M_3x3` (instead of constructing it with padding and an addition), it throws the following error which states that outputs of pytorch and torch scripts models are not equivalent:

```
  /home/jcmaxwell/miniconda2/envs/pytorch-1.0-stable/lib/python3.7/site-packages/torch/jit/__init__.py:642: TracerWarning: Output nr 1. of the traced function does not match the corresponding output of the Python function. Detailed error:
  Not within tolerance rtol=1e-05 atol=1e-05 at input[0, 0, 50, 34] (4.5528998374938965 vs. 0.0) and 49150 other locations (99.00%)
    _check_trace([example_inputs], func, executor_options, module, check_tolerance, _force_outplace)
```

This error directs us a simpler solution :+1: Thanks!